src/pages/UnlockedWallet/AddressesPage/AddressGridRow.tsx

```diff
@@ -0,0 +1,203 @@
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import AddressBadge from '@/components/AddressBadge'
+import AddressColorIndicator from '@/components/AddressColorIndicator'
+import Amount from '@/components/Amount'
+import SkeletonLoader from '@/components/SkeletonLoader'
+import { useAppSelector } from '@/hooks/redux'
+import AddressDetailsModal from '@/modals/AddressDetailsModal'
+import ModalPortal from '@/modals/ModalPortal'
+import {
+  selectAddressByHash,
+  selectAddressesAssets,
+  selectIsStateUninitialized
+} from '@/storage/addresses/addressesSelectors'
+import { selectIsLoadingAssetsInfo } from '@/storage/assets/assetsSelectors'
+import { useGetPriceQuery } from '@/storage/assets/priceApiSlice'
+import { AddressHash } from '@/types/addresses'
+import { currencies } from '@/utils/currencies'
+import { useWindowResize } from '@/utils/hooks'
+
+interface AddressGridRowProps {
+  addressHash: AddressHash
+  className?: string
+}
+
+const assetSymbolsRowHeight = 34
+
+const AddressGridRow = ({ addressHash, className }: AddressGridRowProps) => {
+  const { t } = useTranslation()
+  const address = useAppSelector((state) => selectAddressByHash(state, addressHash))
+  const assets = useAppSelector((state) => selectAddressesAssets(state, address?.hash ? [address.hash] : undefined))
+  const stateUninitialized = useAppSelector(selectIsStateUninitialized)
+  const isLoadingAssetsInfo = useAppSelector(selectIsLoadingAssetsInfo)
+  const { data: price, isLoading: isPriceLoading } = useGetPriceQuery(currencies.USD.ticker)
+
+  const [isAddressDetailsModalOpen, setIsAddressDetailsModalOpen] = useState(false)
+
+  const assetsWithBalance = assets.filter((asset) => asset.balance > 0)
+  const knownAssets = assetsWithBalance.filter((asset) => !!asset.symbol)
+  const [assetSymbolTexts, setAssetSymbolTexts] = useState(map(knownAssets, 'symbol'))
+
+  const assetsCellRef = useRef<HTMLDivElement>(null)
+
+  const ensureAssetsDisplayedInOnlyOneRow = useCallback(() => {
+    if (!assetsCellRef.current) return
+
+    if (assetsCellRef.current.offsetHeight > assetSymbolsRowHeight) {
+      const newArray = [...assetSymbolTexts]
+      const lastItem = newArray.pop()
+
+      if (lastItem?.startsWith('+')) {
+        const num = parseInt(lastItem.replace('+', ''))
+        newArray.pop()
+        newArray.push(`+${num + 1}`)
+      }
+      setAssetSymbolTexts(newArray)
+    } else if (!assetSymbolTexts.at(-1)?.startsWith('+') && assetSymbolTexts.length < assetsWithBalance.length) {
+      const newArray = [...assetSymbolTexts]
+      newArray.push(`+${assetsWithBalance.length - assetSymbolTexts.length}`)
+      setAssetSymbolTexts(newArray)
+    }
+  }, [assetSymbolTexts, assetsWithBalance.length])
+
+  useEffect(() => {
+    ensureAssetsDisplayedInOnlyOneRow()
+  }, [ensureAssetsDisplayedInOnlyOneRow])
+
+  useWindowResize(ensureAssetsDisplayedInOnlyOneRow)
```

Comment:
This piece of code is responsible for this. It works when resizing the window as well, but only when scaling down 😅 

<img width="575" alt="image" src="https://user-images.githubusercontent.com/1579899/226979325-58aafea0-3774-4e9c-8f35-09c765cc972d.png">

Reply:
it now works on all window resize! I wonder if you could give me a hand with framer motion so that we don't have that ugly glitch artifact.

Reply:
Another idea I had in mind is to create an invisible clone of the `AssetSymbols` component instance and position it `absolute` so that it's over the visible one. First do the "add/remove" operation on the invisible component and after a short timeout, based on the content of the invisible container perform the same operation to the visible one. Hard to explain 😅

Reply:
Soooo... After playing a bit with this, I decided to drastically simplify it 🙃 

New design is also more consistent with other of our UIs (tx list, explorer). I really hope you don't mind!
We will probably be able to reuse your logic somewhere else.